### ISSUE TITLE
lx: panic/deadlock possibly related to epoll/signalfd

### ISSUE THREAD

**Issue by wiedi** [joyent/illumos-joyent#120]
Just had a panic:

```
> ::status
debugging crash dump vmcore.0 (64-bit) from azrael
operating system: 5.11 joyent_20160603T232338Z (i86pc)
image uuid: (not set)
panic message: Deadlock: cycle in blocking chain
dump content: kernel pages only
> $C
ffffff005c2f9850 vpanic()
ffffff005c2f98f0 0xfffffffffbb0cd3a()
ffffff005c2f9960 mutex_vector_enter+0x3a3(ffffff1328641480)
ffffff005c2f99e0 signalfd_poll+0xd6(12500000001, 1, 0, ffffff005c2f9b4e, ffffff005c2f9b40)
ffffff005c2f9a20 cdev_poll+0x38(12500000001, 1, 0, ffffff005c2f9b4e, ffffff005c2f9b40)
ffffff005c2f9a60 spec_poll+0x70(ffffff139a8e3480, 1, 0, ffffff005c2f9b4e, ffffff005c2f9b40, 0)
ffffff005c2f9ad0 fop_poll+0x3c(ffffff139a8e3480, 1, 0, ffffff005c2f9b4e, ffffff005c2f9b40, 0)
ffffff005c2f9bb0 dp_pcache_poll+0x18a(ffffff134e0e33f8, ffffff2a2739a840, ffffff2a26677af0, 12, ffffff005c2f9c4c)
ffffff005c2f9cb0 dpioctl+0x4a5(2c00000004, d001, ffffff005c2f9df0, 80200003, ffffff139e93ca28, ffffff005c2f9e1c)
ffffff005c2f9cf0 cdev_ioctl+0x39(2c00000004, d001, ffffff005c2f9df0, 80200003, ffffff139e93ca28, ffffff005c2f9e1c)
ffffff005c2f9d40 spec_ioctl+0x60(ffffff2a54e40c80, d001, ffffff005c2f9df0, 80200003, ffffff139e93ca28, ffffff005c2f9e1c, 0)
ffffff005c2f9dd0 fop_ioctl+0x55(ffffff2a54e40c80, d001, ffffff005c2f9df0, 80200003, ffffff139e93ca28, ffffff005c2f9e1c, 0)
ffffff005c2f9e70 lx_epoll_wait+0xa2(2, 7fffffeff060, 12, ffffffff)
ffffff005c2f9ef0 lx_syscall_enter+0x16f()
ffffff005c2f9f10 sys_syscall+0x16c()

```

If interested I can see how to make the dump available.

**Comment by jjelinek** [comment]
Sebastian,

Yes, if there is any way you can provide the dump then that would be a big
help for us debugging this.

Thanks,
Jerry

**Comment by wiedi** [comment]
I've sent a url in direct mail - let me know if there is more I can do to help

**Comment by jjelinek** [comment]
A fix was pushed last week as 3112412.

**Comment by wiedi** [comment]
Thanks!
